shoulder/gadget/header_depends.py

```diff
@@ -32,6 +34,8 @@ def header_depends(decorated):
 
     def header_depends_decorator(generator, outfile, *args, **kwargs):
         outfile.write("#include <stdint.h>\n")
+        if config.encoded_functions:
```

Comment:
From your comment above: I think this is fine for now, I bet we're going to end up changing how these two files interact at some point anyways, so I like keeping it simple for now